components/docker-up/docker-up/main.go

```diff
@@ -161,10 +175,91 @@ func runWithinNetns() (err error) {
 	return nil
 }
 
+var allowedDockerArgs = map[string]string{
+	"remap-user": "userns-remap",
+}
+
+func setUserArgs(args []string) ([]string, error) {
```

Comment:
This function seemed simpler if it was dedicated to crate args from env vars from the user that were converted to dockerd args. In other words, instead of doing both generation and addition, why not just concentrate on generation?
```suggestion
func userArgs() ([]string, error) {
	args := []string{}
```

And,
```go
	uargs, err := userArgs()
	if err != nil {
		return xerrors.Errorf("cannot add user supplied docker args: %w", err)
	}
	args = append(args, uargs...)
```